### PR TITLE
fix: Remove Auto layout feature flag dependency.

### LINKED ISSUE #24848: [Bug]: Auto-Layout Display Inconsistency When User Is Logged In vs. Not Logged In
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When auto-layout is enabled, there is a discrepancy between how the app is displayed when a user is signed in and when no user is signed in. The difference is more obvious when using a mobile view.
[Discord thread
](https://discord.com/channels/725602949748752515/1122722062587404420)
Anonymous user:

image

Logged-in user:

image

### Steps To Reproduce

Open the attached sample app in a browser where a user is logged in and also open this app on an incognito browser (same screen size).

[AutoLayout_Discrepancy.zip](https://github.com/appsmithorg/appsmith/files/11871015/AutoLayout_Discrepancy.zip)

### Public Sample App

https://app.appsmith.com/app/autolayout-discrepancy/page1-6499ade85148951fb1edb207

### Environment

Production

### Issue video log

_No response_

### Version

Cloud/Self-Hosted v1.9.24

Front logo [Front conversations](https://app.frontapp.com/open/top_kc7po)

### PR DESCRIPTION
**Pull request by marks0351**
> Pull Request Template
>
> Use this template to quickly create a well written pull request. Delete all quotes before creating the pull request.
>
## Description

Removing Feature flag requirement for auto layout since its been in prod for a while and also feature flags are unavailable for public apps.
>
>
#### PR fixes following issue(s)
Fixes #24848
> if no issue exists, please create an issue and ask the maintainers about this first
>
>
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update
>
>
>
- [ ] Manual
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Review by hetunandu — DISMISSED** [review]

File: app/client/cypress/e2e/Regression/ClientSide/MobileResponsiveTests/AutoFillWidgets_Reflow_spec.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      .invoke("css", "flex-wrap")
+      .should("eq", "wrap");
+  });
+  it("2. Auto Layout Reflow should work in public apps as well", () => {
+    let currentUrl = "";
+    cy.url().then((url) => {
+      currentUrl = url;
+      cy.log(currentUrl);
+    });
+    _.homePage.Signout(false).then(() => {
```
Comment (marks0351) [inline app/client/cypress/e2e/Regression/ClientSide/MobileResponsiveTests/AutoFillWidgets_Reflow_spec.ts:35]: @Vijetha-Kaja I have returned the resultant promise of Signout function to wait for its steps to finish and then perform the other steps.

File: app/client/cypress/e2e/Regression/ClientSide/MobileResponsiveTests/AutoFillWidgets_Reflow_spec.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  before(() => {
+    _.autoLayout.ConvertToAutoLayoutAndVerify(false);
+    _.agHelper.Sleep(2000);
+    cy.fixture("autoLayoutReflow").then((val) => {
+      _.agHelper.AddDsl(val);
+    });
+  });
+
+  it("1. Containers Should Reflow in smaller viewports", () => {
+    cy.get(HomePage.shareApp).click();
```
Comment (Vijetha-Kaja) [inline app/client/cypress/e2e/Regression/ClientSide/MobileResponsiveTests/AutoFillWidgets_Reflow_spec.ts:14]: Please use typescript alternatives wherever possible in this spec. Above can be replaced with agHelper.GetNClick(HomePage.shareApp)

File: app/client/cypress/e2e/Regression/ClientSide/MobileResponsiveTests/AutoFillWidgets_Reflow_spec.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      .should("eq", "wrap");
+  });
+  it("2. Auto Layout Reflow should work in public apps as well", () => {
+    let currentUrl = "";
+    cy.url().then((url) => {
+      currentUrl = url;
+      cy.log(currentUrl);
+    });
+    _.homePage.Signout(false).then(() => {
+      cy.visit(currentUrl);
```
Comment (Vijetha-Kaja) [inline app/client/cypress/e2e/Regression/ClientSide/MobileResponsiveTests/AutoFillWidgets_Reflow_spec.ts:36]: Same as above

File: app/client/cypress/e2e/Regression/ClientSide/MobileResponsiveTests/AutoFillWidgets_Reflow_spec.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+      // wait for page render
+      cy.wait("@getPagesForViewApp").should(
+        "have.nested.property",
+        "response.body.responseMeta.status",
+        200,
+      );
+      _.agHelper
+        .GetElement(".auto-layout-layer-0-0")
+        .invoke("css", "flex-wrap")
+        .should("eq", "nowrap");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/MobileResponsiveTests/AutoFillWidgets_Reflow_spec.ts:46]: Can we use `agHelper.AssertAttribute` here?

File: app/client/cypress/e2e/Regression/ClientSide/MobileResponsiveTests/AutoFillWidgets_Reflow_spec.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
+import * as _ from "../../../../support/Objects/ObjectsCore";
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/MobileResponsiveTests/AutoFillWidgets_Reflow_spec.ts:1]: Import only needed Objects from ObjectsCore.

File: app/client/cypress/e2e/Regression/ClientSide/MobileResponsiveTests/AutoFillWidgets_Reflow_spec.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  });
+
+  it("1. Containers Should Reflow in smaller viewports", () => {
+    cy.get(HomePage.shareApp).click();
+    //@ts-expect-error no type access
+    cy.enablePublicAccess(true);
+    _.deployMode.DeployApp();
+
+    _.agHelper
+      .GetElement(".auto-layout-layer-0-0")
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/MobileResponsiveTests/AutoFillWidgets_Reflow_spec.ts:20]: No direct locators in spec files.

**Comment by kamakshibhat-appsmith** [comment]
Tested and looks good .
- The auto layout works fine for unauthenticated users

**Review by Vijetha-Kaja — APPROVED** [review]

**Pull request merged**